src/services/StorageService.ts

```diff
@@ -0,0 +1,28 @@
+export class StorageService {
+  static getLatencies(env: string): Record<string | number, number> {
+    if (env === "browser") {
+      return JSON.parse(localStorage.getItem("rpcLatencies") || "{}");
```

Comment:
So it's both, I felt that would be better. The invoking logic doesn't need to do anything besides initialize and the rest is taken care of, it saves having to repeat code across multiple UI's and for node envs the state is still saved onto the object

what would you suggest, that I remove this and have each UI handle it's own storage?

I'd need to expose the runtimeRPCSs and latencies, then the UI would pull those from the object after init and then save them into local. Anytime getFastestProvider is called the UI will pull from local, pass them into the handler then save the output to local again?

Reply:
You can consider decoupling the logic if you don't want each UI to handle it redundantly. Just have the localStorage logic in a separate method. 

However I would argue that it may not be necessary to use localStorage at all. If we do a Promise.race once upon page load  every time that's probably not bad and will accommodate for changing rpc conditions on a sporadic basis. 

Since all the RPC endpoints fire off in parallel we shouldn't have any bottle necks initializing like this.

Reply:
Like a boolean passed into init enabling local storage handling automatically then methods that just set/get local storage?

The only problem I think is, with every page refresh the class is re-initialized and because the class state of latencies is <= 1 then it'll always race all RPCs. On init I pull the latencies and cache refresh count from localStorage and with passing the refresh arg in on init the UI is able to maintain a cache, so I think local is needed unless I'm not seeing something

We already initialize like you say essence just with the cached rpcs (unless refreshing the list), I can't think how we'd persist state across page visits without using local

Reply:
Yes I just think its not really necessary and adds complexity to making this polymorphic, or also being able to run in node.